clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -1031,6 +1031,8 @@ def err_mainlike_template_decl : Error<"%0 cannot be a template">;
 def err_main_returns_nonint : Error<"'main' must return 'int'">;
 def ext_main_returns_nonint : ExtWarn<"return type of 'main' is not 'int'">,
     InGroup<MainReturnType>;
+def ext_main_no_return : Extension<"non-void 'main' function does not return a value">,
```

Comment:
This isn't really explaining much to the user, just describing what the code does. I think better wording might be: `implicit '0' return value from 'main' is a C99 extension`.